### ISSUE TITLE
[Bug?]: `GET` is not a function when used in route load function

### ISSUE THREAD

**Issue by doeixd** [solidjs/solid-start#1362]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Current behavior 😯

Hello. When attempting to use `GET` from `@solidjs/start`  in a route load function I get the following error:

```
1:54:28 PM [vite] Error when evaluating SSR module #vinxi/handler/ssr: failed to import "/src/entry-server.tsx"
|- TypeError: __vite_ssr_import_6__.GET is not a function
    at eval (C:/Users/Pglenn/Desktop/convert/Final/amring/src/routes/index.tsx?pick=route:38:50)
    at async instantiateModule (file:///C:/Users/Pglenn/Desktop/convert/Final/amring/node_modules/vite/dist/node/chunks/dep-jDlpJiMN.js:54725:9)

[h3] [unhandled] TypeError: __vite_ssr_import_6__.GET is not a function
    at eval (C:/Users/Pglenn/Desktop/convert/Final/amring/src/routes/index.tsx?pick=route:38:50)
    at async instantiateModule (file:///C:/Users/Pglenn/Desktop/convert/Final/amring/node_modules/vite/dist/node/chunks/dep-jDlpJiMN.js:54725:9) {
  cause: TypeError: __vite_ssr_import_6__.GET is not a function
      at eval (C:/Users/Pglenn/Desktop/convert/Final/amring/src/routes/index.tsx?pick=route:38:50)
      at async instantiateModule (file:///C:/Users/Pglenn/Desktop/convert/Final/amring/node_modules/vite/dist/node/chunks/dep-jDlpJiMN.js:54725:9),
  statusCode: 500,
  fatal: false,
  unhandled: true,
  statusMessage: undefined,
  data: undefined
}
```

If I'm using this wrong please let me know.

### Steps to reproduce 🕹

Steps:

1. Use a `GET` function inside a route load function

[Reproduction link](https://codesandbox.io/p/devbox/peaceful-browser-k2frpy?layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clt5rte0d00063j6ozaygsf23%2522%252C%2522sizes%2522%253A%255B47.47610988180357%252C52.52389011819643%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clt5rte0d00023j6o4d82d84g%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clt5rte0d00043j6okhwlml0i%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clt5rte0d00053j6o0ucgm672%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B54.4727440211251%252C45.5272559788749%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clt5rte0d00023j6o4d82d84g%2522%253A%257B%2522id%2522%253A%2522clt5rte0d00023j6o4d82d84g%2522%252C%2522tabs%2522%253A%255B%255D%257D%252C%2522clt5rte0d00053j6o0ucgm672%2522%253A%257B%2522id%2522%253A%2522clt5rte0d00053j6o0ucgm672%2522%252C%2522tabs%2522%253A%255B%257B%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A3000%252C%2522id%2522%253A%2522clt5s21r600b33j6op6rtybq2%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522path%2522%253A%2522%2522%257D%255D%252C%2522activeTabId%2522%253A%2522clt5s21r600b33j6op6rtybq2%2522%257D%252C%2522clt5rte0d00043j6okhwlml0i%2522%253A%257B%2522id%2522%253A%2522clt5rte0d00043j6okhwlml0i%2522%252C%2522activeTabId%2522%253A%2522cltdb30qu00e73j6pg53r6509%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clt5rtq9j003a3j6oabyyr4yl%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TERMINAL%2522%252C%2522shellId%2522%253A%2522clt5rtqcp013dd9g6gyn95i07%2522%257D%252C%257B%2522id%2522%253A%2522cltdb30qu00e73j6pg53r6509%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TERMINAL%2522%252C%2522shellId%2522%253A%2522cltdb30w00016dai0b9tt8n8n%2522%257D%255D%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D)

### Your environment 🌎

```shell
Windows 11
```

**Comment by ryansolid** [comment]
This suggests a circular dependency problem... It could technically be a compiler issue but since I think `cache` works inline it probably isn't. This is our bug not yours, just need to figure out why and see if it is fixable without changing where imports come from.

**Comment by ryansolid** [comment]
Fixed in 0.7.0 we had to move the import ("@solidjs/start/router") but it should work now.
